docs/DeveloperGuide.md

```diff
@@ -6,7 +6,154 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+Here is a quick overview of the design of the application:
+
+![](diagrams/class/all.png)
+
+Note that the diagram does not include all entities, controllers, and managers
+present in the programme. It is only meant to give a quick overview of the project structure.
+
+The later parts of this guide looks into detail each specific part of the programme.
+
+### Controllers
+A `Controller` is defined as the user-facing part of the programme. It is in charge of getting
+input from the user, as well as managing the user experience and interface, such as printing to the console.
+
+All controllers are derived from the base abstract class `Controller`. This base class acts as a mini-framework, and
+includes the base implementation for the input-loop. In this way, there is no need to rewrite input handling for all 
+extended controllers. 
+
+To create a new `Controller`, the new class simply extends from the `Controller` class. The `Controller` class is then provided
+a list of strings indicating the choices available to the user. **Note that the first choice is always to exit the current controller.** 
+The user will then select these choices using integers (the implementation is handled by the base class, including
+input loops). The new class then implements this abstract method at the least:
+- `protected abstract boolean optionSwitcher(int choice) throws OperationTerminationException`
+  - This method is responsible for selecting the corresponding action to the user's selected choice.
+  For more information and examples, refer to the `Controller` class' documentation, and `MainController` class for an
+  example implementation.
+
+Here is the typical design for the Controller:
+
+![](diagrams/class/controller.png)
```

Comment:
![image](https://user-images.githubusercontent.com/53790951/161183806-b1d87c7e-610a-4967-8e22-05b6f8f38242.png)

Good effort in giving more details about the class! It might be better to explicitly show the abstract methods in the diagram with a {abstract} at the end